### PR TITLE
Update DPP to `0.9-dev`

### PR DESCRIPTION
**Pull request by jawid-h**

### CODE REVIEW

File: lib/grpcServer/handlers/applyStateTransitionHandlerFactory.js [hunk 1]
```diff
@@ -40,15 +40,10 @@ module.exports = function applyStateTransitionHandlerFactory(
...
-
-    if (stPacketBinaryArray === undefined) {
-      throw new InvalidArgumentGrpcError('stateTransitionPacket is not specified');
-    }
+    const stateTransitionBinaryArray = request.getStateTransition();
 
-    if (stHeaderBinaryArray === undefined) {
-      throw new InvalidArgumentGrpcError('stateTransitionHeader is not specified');
+    if (stateTransitionBinaryArray === undefined) {
+      throw new InvalidArgumentGrpcError('stateTransition is not specified');
```
Comment (shumkov) [inline lib/grpcServer/handlers/applyStateTransitionHandlerFactory.js]: State Transition

File: lib/grpcServer/handlers/applyStateTransitionHandlerFactory.js [hunk 2]
```diff
@@ -59,39 +54,25 @@ module.exports = function applyStateTransitionHandlerFactory(
       throw new InvalidArgumentGrpcError('blockHash is not specified');
     }
 
-    let stPacket;
+    let stateTransition;
 
     try {
-      const stPacketHex = Buffer.from(stPacketBinaryArray).toString('hex');
+      const stateTransitionHex = Buffer.from(stateTransitionBinaryArray).toString('hex');
```
Comment (shumkov) [inline lib/grpcServer/handlers/applyStateTransitionHandlerFactory.js]: We encode STs with cbor so just back to Buffer.

File: lib/stateView/applyStateTransitionFactory.js [hunk 3]
```diff
@@ -11,50 +13,53 @@ function applyStateTransitionFactory(
...
-   * @param {StateTransition} stHeader
+   * @param {AbstractStateTransition} stateTransition
    * @param {string} blockHash
    * @param {number} blockHeight
    * @param {MongoDBTransaction} transaction
    * @returns {Promise<Object>}
    */
-  async function applyStateTransition(stPacket, stHeader, blockHash, blockHeight, transaction) {
-    if (stPacket.getContract()) {
+  async function applyStateTransition(stateTransition, blockHash, blockHeight, transaction) {
```
Comment (shumkov) [inline lib/stateView/applyStateTransitionFactory.js]: Rename `transaction` to `stateViewTransaction` in order to remove ambiguity.

File: lib/stateView/applyStateTransitionFactory.js [hunk 4]
```diff
@@ -11,50 +13,53 @@ function applyStateTransitionFactory(
...
+    const stateTransitionType = stateTransition.getType();
+
+    if (stateTransitionType === stateTransitionTypes.DATA_CONTRACT) {
```
Comment (shumkov) [inline lib/stateView/applyStateTransitionFactory.js]: `switch` looks more suitable here, isn't?

```js
switch (stateTransition.getType()) {
   case stateTransitionTypes.DATA_CONTRACT:
   ...
}
```

File: lib/stateView/contract/SVContract.js [hunk 5]
```diff
@@ -1,52 +1,37 @@
...
+    return this.contract.getId();
   }
 
   /**
    * Get Contract
    *
-   * @return {Contract}
+   * @return {DataContract}
    */
   getContract() {
```
Comment (shumkov) [inline lib/stateView/contract/SVContract.js]: getDataContract

File: lib/stateView/contract/SVContract.js [hunk 6]
```diff
@@ -1,52 +1,37 @@
...
-  }
-
-  /**
-   * Get user ID
-   *
-   * @return {string}
-   */
-  getUserId() {
-    return this.userId;
+    return this.contract.getId();
```
Comment (shumkov) [inline lib/stateView/contract/SVContract.js]:
```js
this.getDataContact().getId()
```

File: lib/stateView/contract/SVContractMongoDbRepository.js [hunk 7]
```diff
@@ -114,7 +114,7 @@ class SVContractMongoDbRepository {
       svContract.getContract().serialize(),
     );
 
-    const filter = { _id: svContract.getContractId() };
+    const filter = { _id: svContract.getContract().getId() };
```
Comment (shumkov) [inline lib/stateView/contract/SVContractMongoDbRepository.js]: `getId()` ?

File: lib/stateView/contract/createContractDatabaseFactory.js [hunk 8]
```diff
@@ -13,7 +13,7 @@ function createContractDatabaseFactory(createSVDocumentRepository) {
    */
   async function createContractDatabase(svContract) {
     const documents = svContract.getContract().getDocuments();
-    const contractId = svContract.getContractId();
+    const contractId = svContract.getContract().getId();
```
Comment (shumkov) [inline lib/stateView/contract/createContractDatabaseFactory.js]: svContract.getId()

File: lib/stateView/contract/removeContractDatabaseFactory.js [hunk 9]
```diff
@@ -13,7 +13,7 @@ function removeContractDatabaseFactory(createSVDocumentRepository) {
    */
   async function removeContractDatabase(svContract) {
     const documents = svContract.getContract().getDocuments();
-    const contractId = svContract.getContractId();
+    const contractId = svContract.getContract().getId();
```
Comment (shumkov) [inline lib/stateView/contract/removeContractDatabaseFactory.js]: `svContract.getId();`

File: lib/stateView/contract/SVContract.js (hunk 6 above)
Comment (shumkov) [reply lib/stateView/contract/SVContract.js]: Actually, we could convert to base58 here.

File: lib/stateView/document/updateSVDocumentFactory.js [hunk 10]
```diff
@@ -6,17 +6,18 @@ function updateSVDocumentFactory(createSVDocumentRepository) {
...
-   * @param {string} userId
-   * @param {Reference} reference
    * @param {Document} document
+   * @param {Reference} reference
    * @param {MongoDBTransaction} transaction
    * @returns {Promise<void>}
    */
-  async function updateSVDocument(contractId, userId, reference, document, transaction) {
-    const svDocumentRepository = createSVDocumentRepository(contractId, document.getType());
+  async function updateSVDocument(document, reference, transaction) {
```
Comment (shumkov) [inline lib/stateView/document/updateSVDocumentFactory.js]: stateViewTransaction

File: test/integration/stateView/contract/updateSVContractFactory.spec.js [hunk 11]
```diff
@@ -31,34 +30,29 @@ describe('updateSVContractFactory', () => {
     svContract = getSVContractFixture();
     contract = svContract.getContract();
 
-    contractId = svContract.getContractId();
-    userId = svContract.getUserId();
+    contractId = svContract.getContract().getId();
```
Comment (shumkov) [inline test/integration/stateView/contract/updateSVContractFactory.spec.js]: svContract.getId()

File: test/unit/stateView/contract/SVContract.spec.js [hunk 12]
```diff
@@ -41,14 +35,6 @@ describe('SVContract', () => {
...
 
-  describe('#getUserId', () => {
-    it('should return user ID', () => {
-      const result = svContract.getUserId();
-
-      expect(result).to.equal(userId);
-    });
-  });
-
   describe('#getContract', () => {
```
Comment (shumkov) [inline test/unit/stateView/contract/SVContract.spec.js]: getDataContract

File: test/integration/stateView/document/mongoDbRepository/SVDocumentMongoDbRepository.spec.js [hunk 13]
```diff
@@ -668,28 +667,7 @@ describe('SVDocumentMongoDbRepository', function main() {
...
-        userId: svDocument.getUserId(),
-        stReference: {
-          blockHash: currentReference.getBlockHash(),
-          blockHeight: currentReference.getBlockHeight(),
-          stHeaderHash: currentReference.getSTHash(),
-          stPacketHash: currentReference.getSTPacketHash(),
-        },
-      });
-    });
+    it('should return SVDocument with Document having proper $meta');
```
Comment (shumkov) [inline test/integration/stateView/document/mongoDbRepository/SVDocumentMongoDbRepository.spec.js]: We don't have meta anymore.

File: test/integration/stateView/contract/fetchContractFactory.spec.js [hunk 14]
```diff
@@ -26,7 +26,7 @@ describe('fetchContractFactory', () => {
 
     await svContractMongoDbRepository.store(svContract);
 
-    const contractData = await fetchContract(svContract.getContractId());
+    const contractData = await fetchContract(svContract.getContract().getId());
```
Comment (shumkov) [inline test/integration/stateView/contract/fetchContractFactory.spec.js]: svContract.getId()

File: test/integration/stateView/contract/SVContractMongoDbRepository.spec.js [hunk 15]
```diff
@@ -87,7 +87,7 @@ describe('SVContractMongoDbRepository', () => {
     await svContractRepository.store(svContract);
 
     const result = await mongoDatabase.collection('contracts').findOne({
-      _id: svContract.getContractId(),
+      _id: svContract.getContract().getId(),
```
Comment (shumkov) [inline test/integration/stateView/contract/SVContractMongoDbRepository.spec.js]: svContract.getId()

File: lib/stateView/applyStateTransitionFactory.js [hunk 16]
```diff
@@ -11,53 +13,62 @@ function applyStateTransitionFactory(
...
+            stateViewTransaction,
+          );
+        }
+
+        return {};
+      }
+
+      default:
+        return {};
+    }
```
Comment (shumkov) [inline lib/stateView/applyStateTransitionFactory.js]: Let's set `result` variable and return once.

File: test/integration/stateView/applyStateTransitionFactory.spec.js [hunk 17]
```diff
@@ -51,7 +54,7 @@ describe.skip('applyStateTransitionFactory', () => {
       null,
     );
 
-    dpp.setDataProvider(dataProvider);
+    dpp.dataProvider = dataProvider;
```
Comment (shumkov) [inline test/integration/stateView/applyStateTransitionFactory.spec.js]: why not just pass into the constructor as an option?

Comment (jawid-h) [reply test/integration/stateView/applyStateTransitionFactory.spec.js]: `DataProvider` depends on `fetchContract`, which depends on `svContractMongoDbRepository`, which depends on `dpp`. Cyclic dependency.

File: lib/stateView/applyStateTransitionFactory.js [hunk 18]
```diff
@@ -11,53 +13,68 @@ function applyStateTransitionFactory(
...
+      }
 
-      await updateSVDocument(
-        stPacket.getContractId(),
-        stHeader.extraPayload.regTxId,
-        reference,
-        document,
-        transaction,
-      );
+      default:
```
Comment (shumkov) [inline lib/stateView/applyStateTransitionFactory.js]: It's unnecessary.

File: test/integration/stateView/applyStateTransitionFactory.spec.js (hunk 17 above)
Comment (shumkov) [reply test/integration/stateView/applyStateTransitionFactory.spec.js]: This won't work properly. We intentionally removed `setDataProvider`. You can pass a Data Provider only in the constructor. Let's just create one more DPP instance then in order to solve this cyclic problem?

Comment (shumkov) [reply test/integration/stateView/applyStateTransitionFactory.spec.js]: By the way, why do we need Data Provider here?

**Pull request merged**